test/e2e/network/netpol/model.go

```diff
@@ -0,0 +1,291 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package netpol
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+// Model defines the namespaces, deployments, services, pods, containers and associated
+// data for network policy test cases and provides the source of truth
+type Model struct {
+	Namespaces    []*Namespace
+	allPodStrings *[]PodString
+	allPods       *[]*Pod
+	// the raw data
+	NamespaceNames []string
+	PodNames       []string
+	Ports          []int32
+	Protocols      []v1.Protocol
+	DNSDomain      string
+}
+
+// NewModel instantiates a model
+func NewModel(namespaces []string, podNames []string, ports []int32, protocols []v1.Protocol, dnsDomain string) *Model {
+	model := &Model{
+		NamespaceNames: namespaces,
+		PodNames:       podNames,
+		Ports:          ports,
+		Protocols:      protocols,
+		DNSDomain:      dnsDomain,
+	}
+	framework.Logf("DnsDomain %v", model.DNSDomain)
+
+	for _, ns := range namespaces {
```

Comment:
A comment will help here, so you pass an array with X namespaces, Y pods, Z ports and W protocols and this will output, IIUIC a model will be created with:
X namespaces and Y pods in each namespace
each pod will have W*Z containers, each container listening in one port/protocol

Reply:
👍

Reply:
done pushing shortly

Reply:
pushedd